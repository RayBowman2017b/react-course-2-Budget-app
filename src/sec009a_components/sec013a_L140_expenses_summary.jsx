

/***************************************************************************

sec013a_L140_expenses_summary.jsx

       //  [ DEF1: SFC_expenses_summary <1>]
       //  [ DEF1: GF_map_state_to_props <1>]

K:\A01_Udemy\C023_Complete_React_Web_Dev\Budget-app\
src\sec009a_components\sec013a_L140_expenses_summary.jsx

import { SFC_expenses_summary } from '../../sec009a_components/sec013a_L140_expenses_summary.jsx';
          //[S07251667|sec009a_SFC_expense_dashboard_page.jsx::sec013a_L140_expenses_summary import-1;]

//  SEC_013 --- 140. Build It: Adding Summary Component 18:39


 ***************************************************************************/

import React from 'react';

import { connect } from 'react-redux';

import numeral from 'numeral';

import MP_SLCT_getVisibleExpenses from "../sec011a_L099_selectors/sec011a_L099_SLCT_expenses.jsx";
   //[ MP_SLCT_getVisibleExpenses asn1;]
                     //[S07251670|sec011a_L099_SLCT_expenses.jsx::sec011a_L099_SLCT_expenses import-4;]

//import MP_selectExpensesTotal from "../../sec011a_L099_selectors/sec013a_L139_SLCT_total_expenses.jsx";
import { MP_selectExpensesTotal } from "../sec011a_L099_selectors/sec013a_L139_SLCT_total_expenses.jsx";
   //[ MP_selectExpensesTotal import-1;]
   //[ MP_selectExpensesTotal xxx]
              //[S07251670|sec013a_L139_SLCT_total_expenses.jsx::sec013a_L139_SLCT_total_expenses import-1;^B]


//==========================================================================
//==========================================================================

   //  [S07251664|A01_DIrectory_01.txt::DRC1: SFC_expenses_summary <1>^B]

   //  [S07251667|sec009a_SFC_expense_dashboard_page.jsx::TPL1: SFC_expenses_summary <1>]

//[ DEF1: SFC_expenses_summary <1>^B]
const SFC_expenses_summary = (P_props) => (
    <div>
  {
    P_props.expenses.length === 0 ? (
        <p>No Expenses</p>
    ) :
    P_props.expenses.length === 1 ? (
        <p>Viewing 1 Expense totalling {P_props.total_expenses}</p>
    ) : <p>Viewing {P_props.expenses.length} Expenses totalling {P_props.total_expenses}</p>
  }
    </div>
);
//                                 {/*//[ P_props.total_expenses exe1;]*/}
//             {/*//[ P_props.expenses exe1;]*/}


//  SEC_011 --- 103. Controlled Inputs for Filters 14:21
//  MP_SLCT_getVisibleExpenses

 //  [S07251664|A01_DIrectory_01.txt::DRC1: XPNSMRY.GF_map_state_to_props <1>^B]

       //[ DEF1: GF_map_state_to_props <1>^B]
const GF_map_state_to_props = (P_state) =>
{
    const expenses = MP_SLCT_getVisibleExpenses
                 //[ MP_SLCT_getVisibleExpenses asn1;^B]
    //[S07251670|sec011a_L099_SLCT_expenses.jsx::EXE1: MP_SLCT_getVisibleExpenses <1>^B]
                                           (P_state.expenses, P_state.filters);
    //[S07251671|sec011a_L099_STR_configure_store.jsx::(P_state.expenses, P_state.filters) xrf2;^B]
                    //[S07251664|A01_DIrectory_01.txt::(P_state.expenses, P_state.filters) drc2;^B]

    const raw_total_expenses = MP_selectExpensesTotal (expenses);
                           //[ MP_selectExpensesTotal import-1;^B]
    //[S07251670|sec013a_L139_SLCT_total_expenses.jsx::EXE1: MP_selectExpensesTotal <1>^B]

    const total_expenses = numeral(raw_total_expenses / 100).format("$0,0.00");

    return {  expenses, total_expenses };
  //[ P_props.expenses exe1;^B]
            //[ P_props.total_expenses exe1;^B]
};

export default connect (GF_map_state_to_props) (SFC_expenses_summary);
