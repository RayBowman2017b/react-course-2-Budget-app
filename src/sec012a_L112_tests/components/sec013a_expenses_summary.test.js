

/***************************************************************************

sec013a_expenses_summary.test.js

K:\A01_Udemy\C023_Complete_React_Web_Dev\Budget-app\
src\sec012a_L112_tests\components\sec013a_expenses_summary.test.js

//  SEC_013 --- 140. Build It: Adding Summary Component 18:39


 ***************************************************************************/



import React from 'react';
import { shallow } from 'enzyme';

import { SFC_expenses_summary } from '../../sec009a_components/sec013a_L140_expenses_summary.jsx';

//import { SFC_expense_list } from "../../sec009a_components/sec011a_L101_expense_list.jsx";
//import MP_FXT_expenses from "../fixtures/sec012a_DATA_expenses.js";


describe ("COMPONENT TEST FOR EXPENSES SUMMARY", () =>  {

    it ('should render ExpenseList with expenses', () =>  {
        const L_wrapper = shallow
                        ( <SFC_expense_list expenses={MP_FXT_expenses} /> );

        expect(L_wrapper).toMatchSnapshot();
    } );

    it ('should render "no Expenses" with an empty list', () =>  {
        const L_wrapper = shallow
                        ( <SFC_expense_list expenses={[]} /> );

        expect(L_wrapper).toMatchSnapshot();
    } );

} );
