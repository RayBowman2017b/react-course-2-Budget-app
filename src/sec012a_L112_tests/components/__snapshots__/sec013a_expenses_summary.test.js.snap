// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT TEST FOR EXPENSES SUMMARY should correctly render expense sunnary with 1 expense 1`] = `
<Connect(SFC_expenses_summary)
  total_expense_amount={321}
  total_expense_count={1}
/>
`;

exports[`COMPONENT TEST FOR EXPENSES SUMMARY should correctly render expense sunnary with multiple expenses 1`] = `
<Connect(SFC_expenses_summary)
  total_expense_amount={321123}
  total_expense_count={11}
/>
`;
